ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3SubsystemXMLPersister.java

```diff
@@ -489,8 +491,10 @@ private void writeStrictMaxPoolConfig(final XMLExtendedStreamWriter writer, fina
 
         writer.writeAttribute(EJB3SubsystemXMLAttribute.NAME.getLocalName(), strictMaxPoolModel.getName());
 
-        StrictMaxPoolResourceDefinition.MAX_POOL_SIZE.marshallAsAttribute(strictMaxPoolModelNode, writer);
-        StrictMaxPoolResourceDefinition.DERIVE_SIZE.marshallAsAttribute(strictMaxPoolModelNode, writer);
+        if (strictMaxPoolModelNode.hasDefined(MAX_POOL_SIZE))
+            StrictMaxPoolResourceDefinition.MAX_POOL_SIZE.marshallAsAttribute(strictMaxPoolModelNode, writer);
+        if (strictMaxPoolModelNode.hasDefined(DERIVE_SIZE))
+            StrictMaxPoolResourceDefinition.DERIVE_SIZE.marshallAsAttribute(strictMaxPoolModelNode, writer);
```

Comment:
AFAICT the changes to this file should not be necessary as the behavior of these AttributeDefinitions is to not write anything if the attribute isn't defined.

Reply:
OK, i'll remove the changes. Thanks.